spark/src/main/scala/ai/chronon/spark/stats/ConsistencyJob.scala

```diff
@@ -88,10 +88,18 @@ class ConsistencyJob(session: SparkSession, joinConf: Join, endDate: String) ext
     if (unfilledRanges.isEmpty) return null
     val allMetrics = unfilledRanges.map { unfilled =>
       val comparisonDf = tableUtils.sql(unfilled.genScanQuery(null, joinConf.metaData.comparisonTable))
+      // External parts / contextual features don't get logged in the online data but do appear in the comparison table.
+      // We need to remove them from the comparison df otherwise the comparison metric computation will fail.
+      // We need to prepend `ext_contextual_` to the feature name since that's how it appears in the comparison df.
+      val externalFeatureColumns = joinConf.getExternalFeatureCols.map(col => s"ext_contextual_$col")
```

Comment:
the column name logic seems wrong. `getExternalFeatureCols` currently has no usages in the codebase and doesn't handle prefix correctly. 

`ext_contextual_` is the prefix for only contextual features which is a special case of external features, the general case external features has prefix based on its source name. 

So instead we should use:

```
Option(joinConf.onlineExternalParts).toSeq.flatMap(_.toScala.map(_.fullName))
```